src/interface_matlab.cpp

```diff
@@ -178,7 +187,11 @@ ExpData *expDataFromMatlabCall(const mxArray *prhs[],
     int ne_mz = 0, nz_mz = 0, ne_sigmaz = 0,
         nz_sigmaz = 0; /* integers with problem dimensionality */
     ExpData *edata = new ExpData(model);
-    if (edata->my.empty() && edata->mz.empty()) {
+    
+    /* Check whether data and events are empty */
+    std::vector<realtype> observedData = edata->getObservedData();
+    std::vector<realtype> observedEvents = edata->getObservedEvents();
+    if (observedData.empty() && observedEvents.empty()) {
         // if my and mz are both empty, no (or empty) data was provided
         // in that case we simply return a nullptr for easier checking.
         delete(edata);
```

Comment:
@FFroehlich: I don't get what is supposed to happen in these lines.  Do we still need that?

Reply:
We should be able to completely remove this as data fields are no longer mandatory.